### PR TITLE
Publish artifacts syntax

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: docs/preview/yamlgettingstarted-publish.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+```yaml
+steps:
+- script: |
+    mkdir layoutDir
+    echo hello > layoutDir/hello.txt
+    echo world > layoutDir/world.txt
+
+- publishArtifact: layout
+  sourceFolder: layoutDir # or $(system.defaultWorkingDirectory)/layoutDir
```
Comment (chrisrpatterson) [inline docs/preview/yamlgettingstarted-publish.md]: how about path instead of sourceFolder

Comment (ericsciple) [reply docs/preview/yamlgettingstarted-publish.md]: oops I originally meant to put `sourcePath`.

I don't mind changing this to simply `path`. My thinking was to make way for `targetPath` for things like file share artifacts, or even the current server artifacts can use a user-specified path to put it under a specific folder. Multiple publish steps can be used today to contribute to a single artifact, if the subpaths are different.

File: docs/preview/yamlgettingstarted-publish.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+In the following example, a custom folder is used to stage the artifact.
+
+```yaml
+steps:
+- script: |
+    mkdir layoutDir
+    echo hello > layoutDir/hello.txt
+    echo world > layoutDir/world.txt
+
+- publishArtifact: layout
```
Comment (ericsciple) [inline docs/preview/yamlgettingstarted-publish.md]: would `publish` be sufficient?

File: docs/preview/yamlgettingstarted-publish.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+In the following example, the first step creates a file under the artifact staging directory.
+
+The `publishArtifact` step uploads the artifact staging directory. When the build completes, the artifact named `drop` can be downloaded from the artfacts tab on the build summary page.
+
+```yaml
+steps:
+- script: echo hello > $(build.artifactStagingDirectory)/hello.txt
+
+- publishArtifact: drop
```
Comment (DavidStaheli) [inline docs/preview/yamlgettingstarted-publish.md]: Perhaps don't require an artifact name and default to "drop" if not specified.  That's what our designer templates do with the PublishBuildArtifacts task.

**Comment by jtpetty** [comment]
@vtbassmatt - Can you take a look at this one?

**Pull request closed without merging**